internal/aws/awsutil/conn.go

```diff
@@ -383,3 +356,51 @@ func getPartition(region string) string {
 	p, _ := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
 	return p.ID()
 }
+
+// order
+// 1. override creds providers
+// 2. shared creds file
+// 3. shared profile
+// explicit keys do not make sense in this context since contrib does not take in explicit keys
+// and cwa does not provide explicit keys to config but come from these methods of getting creds
+// @TODO make this more upstream friendly working in the future
+func customCredentialProvider(cfg *AWSSessionSettings) []credentials.Provider {
```

Comment:
nit: This is a credential provider chain rather than a single custom credential provider. 

```suggestion
func getCredentialProviderChain(cfg *AWSSessionSettings) []credentials.Provider {
```